### PR TITLE
feat: Restrict import media for users that are not allowed to share (AR-3223) (AR-3222)

### PR DESCRIPTION
**Pull request by borichellow**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-3223" title="AR-3223" target="_blank">StoryAR-3223</a>  Restrict Import Media feature to users from team with file sharing restrictions
  </td></table>
  <br />
 

# What's new in this PR?

Show the "FileSharing is Restricted" screen to the users that are trying to share files and are in a team that is not allowed it.
OR are not logged in.
And do not allow them to share anything so by it

### Attachments (Optional)

photo_5289645575765084427_y

photo_5291897375578770077_y

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -892,8 +892,8 @@
     <string name="label_system_message_session_reset">%1$s was unable to decrypt some of your messages but has solved the issue. This affected all conversations you share together.</string>
     <string name="label_manual_migration_title">Migrate from version 3.82.38</string>
     <string name="start_manual_migration">Start migration form version 3.82.38 manually</string>
-    <string name="file_sharing_restricted_dialog_title">File Sharing Restrictions</string>
-    <string name="file_sharing_restricted_dialog_description">You can not share this file because this feature is disabled.</string>
+    <string name="file_sharing_restricted_lear_more_link">https://support.wire.com/hc/en-us/articles/4406404582673-How-can-I-restrict-file-sharing-for-my-team-</string>
```
Comment (gongracr) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="file_sharing_restricted_learn_more_link">https://support.wire.com/hc/en-us/articles/4406404582673-How-can-I-restrict-file-sharing-for-my-team-</string>
```

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1636/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6172418)(6172418)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6172418)(6172418)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1636/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6174149)(6174149)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6174149)(6174149)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt [hunk 2]
```diff
@@ -137,14 +139,22 @@ class FeatureFlagNotificationViewModel @Inject constructor(
         // This function needs to be executed blocking the main thread because otherwise the list of imported assets will not be updated
         // correctly for some strange reason.
         runBlocking {
-            if (checkNumberOfSessions() > 0) {
-                featureFlagState = featureFlagState.copy(showFileSharingRestrictedDialog = featureFlagState.isFileSharingEnabledState)
+            val fileSharingRestrictedWallState = if (checkNumberOfSessions() > 0) {
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt]: Raise PR's not Walls 😢 ❤️ 🧱 🔨

**Review by gongracr — APPROVED** [review]
LGTM 🍬 🍪

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1636/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6174539)(6174539)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6174539)(6174539)-staging-compat.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1636/5/) **failed**.

**Review by MohamadJaara — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-1636/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv4.1.1(6232768)(6232768)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-vv4.1.1(6232768)(6232768)-staging-compat.apk)

**Pull request merged**